vortex-datafusion/src/persistent/format.rs

```diff
@@ -0,0 +1,148 @@
+use std::any::Any;
+use std::sync::Arc;
+
+use arrow_schema::{Schema, SchemaRef};
+use async_trait::async_trait;
+use datafusion::datasource::file_format::file_compression_type::FileCompressionType;
+use datafusion::datasource::file_format::{FileFormat, FilePushdownSupport};
+use datafusion::datasource::physical_plan::{FileScanConfig, FileSinkConfig};
+use datafusion::execution::SessionState;
+use datafusion_common::parsers::CompressionTypeVariant;
+use datafusion_common::stats::Precision;
+use datafusion_common::{not_impl_err, DataFusionError, Result as DFResult, Statistics};
+use datafusion_expr::Expr;
+use datafusion_physical_expr::{LexRequirement, PhysicalExpr};
+use datafusion_physical_plan::metrics::ExecutionPlanMetricsSet;
+use datafusion_physical_plan::ExecutionPlan;
+use object_store::{ObjectMeta, ObjectStore};
+use vortex_array::arrow::infer_schema;
+use vortex_array::Context;
+use vortex_dtype::DType;
+use vortex_file::VORTEX_FILE_EXTENSION;
+use vortex_io::ObjectStoreReadAt;
+
+use super::execution::VortexExec;
+use crate::can_be_pushed_down;
+
+#[derive(Debug, Default)]
+pub struct VortexFormat {
+    context: Arc<Context>,
+}
+
+impl VortexFormat {
+    pub fn new(context: &Context) -> Self {
+        Self {
+            context: Arc::new(context.clone()),
+        }
+    }
+}
+
+#[async_trait]
+impl FileFormat for VortexFormat {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_ext(&self) -> String {
+        VORTEX_FILE_EXTENSION.to_string()
+    }
+
+    fn get_ext_with_compression(
+        &self,
+        file_compression_type: &FileCompressionType,
+    ) -> DFResult<String> {
+        match file_compression_type.get_variant() {
+            CompressionTypeVariant::UNCOMPRESSED => Ok(self.get_ext()),
+            _ => Err(DataFusionError::Internal(
+                "Vortex does not support file level compression.".into(),
+            )),
+        }
+    }
+
+    async fn infer_schema(
```

Comment:
This api is unfortunate, it means you can't cache requests across stats and regular read

Reply:
As long as we have our own internal IO system that doesn't use the externally provided `ObjectStore` instance, I think that's going to happen to some extent anyway.